File: app/src/main/java/ru/aradxxx/ciceronetabssample/tabfragment/TabFragment.java
package ru.aradxxx.ciceronetabssample.tabfragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import ru.aradxxx.ciceronetabssample.R;
import ru.aradxxx.ciceronetabs.NavigationContainer;
import ru.aradxxx.ciceronetabs.TabCicerone;
import ru.aradxxx.ciceronetabs.TabNavigator;
import ru.aradxxx.ciceronetabs.TabRouter;
import ru.aradxxx.ciceronetabssample.MainActivity;
import ru.aradxxx.ciceronetabssample.Tab;
import ru.aradxxx.ciceronetabssample.TabListener;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import dagger.android.support.DaggerFragment;

/**
 * Пример базового класса фрагмента таба.
 */
public abstract class TabFragment extends DaggerFragment
        implements NavigationContainer<TabRouter> {
    @Inject
    TabCicerone<TabRouter> tabCicerone;
    @Nullable
    private TabNavigator<TabRouter> tabNavigator;
    private TabListener tabListener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        Fragment parent = getParentFragment();
        if (parent instanceof TabListener) {
            tabListener = (TabListener) parent;
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_tab, container, false);
    }

    @Override
    public void onResume() {
        super.onResume();
        tabCicerone.cicerone(tab().tag()).getNavigatorHolder().setNavigator(tabNavigator());
        updateContainerTab();
    }

    @Override
    public void onPause() {
        tabCicerone.cicerone(tab().tag()).getNavigatorHolder().removeNavigator();
        super.onPause();
    }

    @NonNull
    @Override
    public TabRouter router() {
        return tabCicerone.router(tab().tag());
    }

    @NonNull
    protected abstract Tab tab();

    private void updateContainerTab() {
        if (tabListener != null) {
            tabListener.tabResumed(tab());
        }
    }

    @NonNull
    private TabNavigator tabNavigator() {
        if (tabNavigator == null) {
            MainActivity activity = (MainActivity) requireActivity();
            tabNavigator = new TabNavigator<>(activity, tabCicerone, getChildFragmentManager(), R.id.fc_container);
        }
        return tabNavigator;
    }
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/tabfragment/TabFragment.kt
package com.github.aradxxx.ciceroneflowssample.tabfragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.github.aradxxx.ciceroneflow.FlowCicerone
import com.github.aradxxx.ciceroneflow.FlowNavigator
import com.github.aradxxx.ciceroneflow.FlowRouter
import com.github.aradxxx.ciceroneflow.NavigationContainer
import dagger.android.support.DaggerFragment
import com.github.aradxxx.ciceroneflowssample.R
import com.github.aradxxx.ciceroneflowssample.Tab
import com.github.aradxxx.ciceroneflowssample.TabListener
import javax.inject.Inject

/**
 * Пример базового класса фрагмента контейнера.
 */
abstract class TabFragment : DaggerFragment(), NavigationContainer<FlowRouter> {
    @Inject
    lateinit var flowCicerone: FlowCicerone<FlowRouter>
    private val flowNavigator by lazy {
        FlowNavigator(
            requireActivity(),
            R.id.fc_container,
            flowCicerone,
            requireActivity().supportFragmentManager,
            childFragmentManager)
    }
    private var tabListener: TabListener? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        val parent = parentFragment
        if (parent is TabListener) {
            tabListener = parent
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_tab, container, false)
    }

    override fun onResume() {
        super.onResume()
        flowCicerone.cicerone(tab().tag()).getNavigatorHolder().setNavigator(flowNavigator)
        updateContainerTab()
    }

    override fun onPause() {
        flowCicerone.cicerone(tab().tag()).getNavigatorHolder().removeNavigator()
        super.onPause()
    }

    override fun router(): FlowRouter {
        return flowCicerone.router(tab().tag())
    }

    protected abstract fun tab(): Tab
    private fun updateContainerTab() {
        tabListener?.tabResumed(tab())
    }
}
